### PR TITLE
Q288 Add uniqueWordAbbreviation.py

### PR DESCRIPTION
**Pull request by peinanteng**

### CODE REVIEW

File: uniqueWordAbbreviation.py [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+        :type word: str
+        :rtype: bool
+        """
+        if len(word) <= 2:
+            newWord = word
+        else:
+            newWord = word[0] + str(len(word)-2) + word[-1]
+        if (word in self.dict and self.summary[newWord] > 1) or (word not in self.dict and newWord in self.summary):
+                return False
+        else:
```
Comment (nkukarl) [inline uniqueWordAbbreviation.py]: `else` not needed

File: uniqueWordAbbreviation.py [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+    def isUnique(self, word):
+        """
+        :type word: str
+        :rtype: bool
+        """
+        if len(word) <= 2:
+            newWord = word
+        else:
+            newWord = word[0] + str(len(word)-2) + word[-1]
+        if (word in self.dict and self.summary[newWord] > 1) or (word not in self.dict and newWord in self.summary):
```
Comment (nkukarl) [inline uniqueWordAbbreviation.py]: Too long

File: uniqueWordAbbreviation.py [hunk 3]
```diff
@@ -0,0 +1,29 @@
+class ValidWordAbbr(object):
+
+    def __init__(self, dictionary):
+        """
+        :type dictionary: List[str]
+        """
+        self.summary = {}
+        self.dict = set(dictionary)
```
Comment (nkukarl) [inline uniqueWordAbbreviation.py]: `dict` is a reserved word

File: uniqueWordAbbreviation.py [hunk 4]
```diff
@@ -0,0 +1,29 @@
...
+        for word in self.dict:
+            if len(word) <= 2:
+                self.summary[word] = self.summary.get(word, 0) + 1
+            else:
+                newWord = word[0] + str(len(word)-2) + word[-1]
+                self.summary[newWord] = self.summary.get(newWord, 0) + 1
+
```
Comment (nkukarl) [inline uniqueWordAbbreviation.py]: One blank line

File: uniqueWordAbbreviation.py [hunk 5]
```diff
@@ -5,8 +5,8 @@ def __init__(self, dictionary):
         :type dictionary: List[str]
         """
         self.summary = {}
```
Comment (nkukarl) [inline uniqueWordAbbreviation.py]: Why do you need `self.summary`?

File: uniqueWordAbbreviation.py [hunk 6]
```diff
@@ -5,8 +5,8 @@ def __init__(self, dictionary):
...
-        self.dict = set(dictionary)
-        for word in self.dict:
+        self.dictionary = set(dictionary)
```
Comment (nkukarl) [inline uniqueWordAbbreviation.py]: Why do you need `self.dictionary`?

File: uniqueWordAbbreviation.py (hunk 5 above)
Comment (peinanteng) [reply uniqueWordAbbreviation.py]: Because the function needs initialisation.

File: uniqueWordAbbreviation.py (hunk 6 above)
Comment (peinanteng) [reply uniqueWordAbbreviation.py]: Because the function needs initialisation.

Comment (nkukarl) [reply uniqueWordAbbreviation.py]: I don't understand, why cannot you just use `dictionary_` and `summary`?

**Pull request merged**